### PR TITLE
Counts for GMCR population

### PR DESCRIPTION
**Pull request by maiparkes58**

### CODE REVIEW

**Review by Georgetilston — APPROVED** [review]
all looks good to me. Good to see that much of it reuses code from the reusable queries!

**Comment by rw251** [comment]
I think this might be missing a new file: `query-patient-lsoa-gm.sql`

I can see a `query-lsoa-gm-lookup.sql` and a `query-patient-lsoa.sql` - but not the one above.

@maiparkes58  can you check that out for me? Thanks.

**Comment by rw251** [comment]
Thanks @maiparkes58. Do you perhaps also need `query-patient-lsoa.sql` ? Some of the subsequent temp tables that are used come from that query.

Just thinking about this some more... I think the home for this should be in the `./projects/Reports/template-sql` directory. You would then rename this file to include the extension `.template.sql`. Then when you run the `generate-sql` script, a sql file for execution, with all the `--> EXECUTE` queries stitched together, would appear in `extraction-sql`. I think when you do this you'll find a few errors if you try to execute the whole file at once (e.g. unknown temp tables / dropping temp tables that don't exist). Let me know once you've done all this and I'll take another look. Thanks again.

**Comment by maiparkes58** [comment]
Hi @rw251. Yeah that is a good approach. I found some bugs by doing that and updated the scripts and then repeated the procedure again and the file worked. Thank you.

File: shared/population-counts [hunk 1]
```diff
@@ -0,0 +1,280 @@
+--┌───────────────────────────┐
```
Comment (rw251) [inline shared/population-counts:1]: This file can be deleted, now that you have the one in the Reports folder.

File: projects/Reports/template-sql/population-counts.template.sql [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+LEFT OUTER JOIN #CCGLookup ccg ON ccg.CcgId = gp.Commissioner;
+
+-- Tidy up - helpful in ensuring the tempdb doesn't run out of space mid-query
+DROP TABLE #PatientPractice;
+DROP TABLE #UnmatchedPatientsForPracticeCode;
+DROP TABLE #CCGLookup;
+
+
+-- Find patients opt out of data sharing (ID, Opt_Out)==============================================================================================================
+-- If patients have a tenancy id of 2 we take this as their most likely GP practice
```
Comment (rw251) [inline projects/Reports/template-sql/population-counts.template.sql:111]: The comments here, and a few below talk about "likely GP practice" - but it's actually "likely opt-out status"

File: projects/Reports/template-sql/population-counts.template.sql [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+DROP TABLE #PatientsGPEvent;
+DROP TABLE #PatientsGPMedication;
+
+
+-- Get information for GMCR population===========================================================================================================================
+-- Today date
+SELECT FORMAT(GETDATE(), 'yyyy-MM-dd');
+
+-- All patients in GMCR
+SELECT COUNT (*) FROM #Population;
```
Comment (rw251) [inline projects/Reports/template-sql/population-counts.template.sql:212]: For all of these `SELECT COUNT(*)` statements you should add a label e.g. `SELECT COUNT(*) AS TotalPopulation`. It just means that when you run them all at once you can see at a glance in the results which each number refers to.

Alterntively you could do something like this to bring them all into a single `SELECT` statement:

```sql
SELECT 
  COUNT(*) AS TotalPopulation,
  SUM(CASE WHEN Tenancy_ID = 2 THEN 1 ELSE 0 END) AS GPRegistered,
  SUM(CASE WHEN Tenancy_ID IS NULL THEN 1 ELSE 0 END) AS NotGPRegistered,
  -- ... more lines here...
FROM #Population
```

**Review by rw251 — COMMENTED** [review]
@maiparkes58 this looks good to me. I've just added a few comments for some small changes, then I can approve.

**Review by rw251 — APPROVED** [review]
Looks good. Thanks for making the changes.

**Pull request merged**